crates/bdk-ldk/src/lib.rs

```diff
@@ -366,8 +374,10 @@ where
             ConfirmationTarget::HighPriority => 1,
         };
 
-        let estimate = self.client.estimate_fee(target_blocks).unwrap_or_default();
-        let sats_per_vbyte = estimate.as_sat_per_vb() as u32;
+        let sats_per_vbyte = self
+            .fee_estimator
+            .estimate_fee_blocking(target_blocks)
+            .unwrap_or_default();
```

Comment:
This would still end up with the default value of u32 `0` if the mempool call fails. I guess thats kind of the same issue we had using the electrum_client as fee estimator. At least I would not use `.unwrap_or_default()`, but rather handle the error and print that we failed to fetch the fee rate from mempool. I also wouldn't return 0 as fee rate as this would always end up with a fee rate of 253 sats.

Instead I'd suggest to return something like `MAX_SATS_PER_V_BYTE`. This is of course not perfect, but I guess a better value than `0`, as judging from the ldk code I think we have better chances of preventing the channel to get force closed. But its definitely not a guarantee.

----

LDK will check on the upper limit like this.

```
let upper_limit = cmp::max(250 * 25,
			fee_estimator.bounded_sat_per_1000_weight(ConfirmationTarget::HighPriority) as u64 * 
```

and for the lower limit like this, with the `feerate_per_kw` being the counterparties proposed feerate.

```
if feerate_per_kw + 250 < lower_limit {
  if let Some(cur_feerate) = cur_feerate_per_kw {
    if feerate_per_kw > cur_feerate {
      log_warn!(logger, "Accepting feerate that may prevent us from closing this channel because it's higher than what we have now. Had {} s/kW, now {} s/kW.", cur_feerate, feerate_per_kw);
      return Ok(());
    }
  }
  return Err(ChannelError::Close(format!("Peer's feerate much too low. Actual: {}. Our expected lower limit: {} (- 250)", feerate_per_kw, lower_limit)));
}
```

Reply:
Thanks for the explanation. I'll see to make this better.

Reply:
> Instead I'd suggest to return something like `MAX_SATS_PER_V_BYTE`. This is of course not perfect, but I guess a better value than `0`, as judging from the ldk code I think we have better chances of preventing the channel to get force closed. But its definitely not a guarantee.

Why do you want to return `MAX_SATS_PER_V_BYTE`? Wouldn't `1 sat/vbyte` be better?

Reply:
> > Instead I'd suggest to return something like `MAX_SATS_PER_V_BYTE`. This is of course not perfect, but I guess a better value than `0`, as judging from the ldk code I think we have better chances of preventing the channel to get force closed. But its definitely not a guarantee.
> 
> Why do you want to return `MAX_SATS_PER_V_BYTE`? Wouldn't `1 sat/vbyte` be better?

If I read the ldk code correctly, the 
- upper limit is at least `250 * 25 = 6_250` , so with a `MAX_SATS_PER_V_BYTE` of `20`, `253 * 20 = 5_060` we should not surpass that threshold.
- lower limit is whatever electrs or now mempool would say. Now if one side would `unwrap_or_default()` to the minimum fee rate of `1 sat/vbyte` we would end up with `253 sats/vbyte`. It's likely that the current fee rate is higher than that.

Hence in order to reduce the risk of a force closure due to a `peer's feerate much too low`, I would pick a larger feerate as we will not reach the upper limit, but have more chances on being larger than the lower limit. But of course its just a "better guess" and not a guarantee, that we will not run into this error again.